libkbfs/folder_update_prepper.go

```diff
@@ -698,7 +698,7 @@ func (fup folderUpdatePrepper) updateResolutionUsageAndPointers(
 	}
 	deletedBlocks := make(map[BlockPointer]bool)
 	for ptr := range toUnref {
-		if ptr == zeroPtr {
+		if ptr == zeroPtr || unmergedChains.doNotUnrefPointers[ptr] {
```

Comment:
If new blocks are created and are only ever buffered in memory, we can't rely on the block server (i.e., the journal) to tell us whether they're unflushed or not.  `SyncAll` needs to indicate those using this new map.